### PR TITLE
StackOverflow debug improvements

### PR DESCRIPTION
**Pull request by noahfalk**
Added some debugging instructions for debugging a StackOverflow on Windows and made some minor improvements to the pre-existing text. Unfortunately the Windows and Linux instructions remain asymmetric because the dump debugging experience for StackOverflow on Windows isn't good (https://github.com/dotnet/diagnostics/issues/4944). In the future we might also want to add instructions for using VSCode on Linux or add more to the lldb debugging instructions but I didn't want to expand the scope too much for now.

Fixes: https://github.com/dotnet/diagnostics/issues/3506

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/diagnostics/debug-stackoverflow.md](https://github.com/dotnet/docs/blob/9649ca2768e0dc71f2d88268ab07be976c3ce6bc/docs/core/diagnostics/debug-stackoverflow.md) | [docs/core/diagnostics/debug-stackoverflow](https://review.learn.microsoft.com/en-us/dotnet/core/diagnostics/debug-stackoverflow?branch=pr-en-us-42673) |

### CODE REVIEW

**Comment by noahfalk** [comment]
@dotnet/dotnet-diag

File: docs/core/diagnostics/debug-stackoverflow.md [hunk 1]
```diff
@@ -39,12 +39,15 @@ Stack overflow.
    <this output repeats many more times>
 ````
 
-Often just seeing this callstack is enough to identify the problematic repeating method and locate the source code logic that is causing the recursive calls. However if the problem is still unclear you can debug further.
+When you see the program exit with output like this you can find the source code for the repeating method(s) and investigate the logic that causes the large number of calls.
```
Comment (tommcdon) [inline docs/core/diagnostics/debug-stackoverflow.md:42]:
```suggestion
When you see the program exit with output like this, you can find the source code for the repeating method(s) and investigate the logic that causes the large number of calls.
```

File: docs/core/diagnostics/debug-stackoverflow.md [hunk 2]
```diff
@@ -103,7 +106,19 @@ Often just seeing this callstack is enough to identify the problematic repeating
     Source file:  /temp/Program.cs @ 9
     ````
 
-5. Go look at the indicated method temp.Program.Main(System.String[]) and source "/temp/Program.cs @ 9" to see if you can figure out what you did wrong. If it still wasn't clear you could use further debugger commands to inspect the current state of the process or add logging in that area of the code.
+5. Go look at the indicated method temp.Program.Main(System.String[]) and source "/temp/Program.cs @ 9" to see if you can figure out what the code is doing wrong. If it still wasn't clear you could use further debugger or [SOS](sos-debugging-extension.md) commands to inspect the process.
```
Comment (tommcdon) [inline docs/core/diagnostics/debug-stackoverflow.md:109]:
```suggestion
5. Go look at the indicated method temp.Program.Main(System.String[]) and source "/temp/Program.cs @ 9" to see if you can figure out what the code is doing wrong. If additional information is needed, you can use further debugger or [SOS](sos-debugging-extension.md) commands to inspect the process.
```

**Review by tommcdon — APPROVED** [review]
Left a few suggestions for your consideration

**Comment by noahfalk** [comment]
Done, thanks Tom!

**Pull request merged**